packages/list-store/src/useListStore.ts

```diff
@@ -0,0 +1,109 @@
+import {create} from 'zustand';
+import {getUrlWithSearchParams} from './search-params';
+import {SortBy, defaultSortBy} from './sort';
+
+export interface Props {
+  totalCount: number;
+  offset: number;
+  limit: number;
+  query: string;
+  sortBy: SortBy;
+  // Setting newDataNeeded to true will trigger a page reload with new search params
+  newDataNeeded: boolean;
+  initialized: boolean;
+  selectedFilters: {[filterKey: string]: string[] | undefined};
+  setSelectedFilters: (key: string, value: string[]) => void;
+  setSortBy: (sortBy: SortBy) => void;
+  setQuery: (query: string) => void;
+  setPage: (direction: 1 | -1) => void;
+  composed: {
```

Comment:
I see where you want to go with this. But, to me, this is also a bit hard to follow. For example: what is the meaning of `composed` - why and when would I need this (if I were to use this store/hook)? Do you see a way to make this clear, or to simplify the implementation?

Reply:
I'm used to using a state manager with the possibility of derived values. (https://mobx-state-tree.js.org/concepts/views). I was trying to do the same, but I can see that this is unclear. I think it is better to move this entire logic to `useSearchParamsUpdate.tsx`. This is the only place this is used, so let's keep the logic all in one place.